pkg/plugin/javascript/dashboard_client.go

```diff
@@ -16,6 +16,15 @@ import (
 	"github.com/vmware-tanzu/octant/internal/octant"
 )
 
+// DashboardMetadataKey is a type used for metadata keys passed by plugins
+type DashboardMetadataKey string
+
+//DashboardMetadataSentinel is a type for CloneKey
+type DashboardMetadataSentinel string
+
+//CloneKey is a dummy key used to clone context.Context
+var CloneKey DashboardMetadataSentinel = ""
```

Comment:
These can go away if we use `WithCancel` to make a copy. Also it avoids some potential confusion when a plugin author tries to send it data and use empty string (which is a valid key) as their key.